### DISCUSSION TITLE
Event correlation on dynamic values

### DISCUSSION THREAD

**Discussion by bakjos** [serverlessworkflow/specification#777] [Q&A]
I have the following use case and am not sure how it can be handled: I’m using a callback state to wait for a human task, the action will create the task and return the task Id and it will wait for an event with the same event type and source, and the task id can be part of the context attributes, but it will wait for any event, not for the one matching the specific task id

**Comment by cdavernas** [comment]
You can achieve that, just like in Synapse, by evaluating your correlations at the time events are consumed, thus allowing you to select the event's data or attributes, or even the current state data.

Note that this implies the runtime must support expression based correlation attribute values, which is an opiniated implementation, as the spec does not define that specific feature.

Furthermore, for complex correlation use cases that imply more than a single correlation key, we would need to define a new key property in the spec at the correlation attribute level, allowing coexistence of different keys (the spec only supports a single one as of now)

**Comment by bakjos** [comment]
Thank you so much for the answer, using an expression as the correlation key makes sense, do you know if there are any plans to have more than one correlation key in the spec in the future?

**Reply by cdavernas** [reply]
I opened https://github.com/serverlessworkflow/specification/issues/680 to address that some time ago.
I'm confident we'll find a satisfying solution for 0.9.

**Reply by brampurnot** [reply]
Any news on this? I'm also interested in it :)

**Reply by cdavernas** [reply]
Yeah, so am I! I'll bring that up in next weekly! You are welcome to join and back up the feature request 😉

**Comment by cdavernas** [comment]
I'm closing the discussion as it applies to a discontinued version of the spec. 

@bakjos @brampurnot I invite you to take a look at the brand new [listen task](https://github.com/serverlessworkflow/specification/blob/main/dsl-reference.md#listen), which allows what was discussed here and much more!

Please feel free to open a new discussion/issue if you have any question!
